### PR TITLE
add changes

### PR DESCRIPTION
**Pull request by piensaenpixel**
@xavijam could you take a look?
drag-ok

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js [hunk 1]
```diff
@@ -1,6 +1,15 @@
 var cdb = require('cartodb-deep-insights.js');
 var template = require('./widget-view.tpl');
 
+var widgetIconTemplateMap = {
+  category: require('./widget-icon-category.tpl'),
+  histogram: require('./widget-icon-histogram.tpl'),
+  formula: require('./widget-icon-formula.tpl')
+  /*,
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js]: Remove this commented code.
BTW, what happened with the torque widget icon?

File: lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js [hunk 2]
```diff
@@ -25,11 +34,22 @@ module.exports = cdb.core.View.extend({
   },
 
   render: function () {
+    var widgetType = this.model.get('type');
     this.$el.html(template({
+      widgetType: this.model.get('type'),
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js]: …
widgetType: widgetType,
…

File: lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js (hunk 1 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js]: Could we create it and leave it blank at least?

File: lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js [hunk 3]
```diff
@@ -25,11 +34,22 @@ module.exports = cdb.core.View.extend({
...
       layerName: this.layer.getName(),
       title: this.model.get('title')
     }));
     this.$el.attr('data-model-cid', this.model.cid);
+
+    var iconTemplate = widgetIconTemplateMap[widgetType];
+
+    if (!iconTemplate) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js]: This should be:
```
if (!iconTemplate) {
   console.log(widgetType + ' widget template not defined');
} else {
   this.$('.js-widgetIcon').append(iconTemplate());
}
…
```

**Comment by xavijam** [comment]
:+1: after green light :)

File: lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js [hunk 4]
```diff
@@ -1,6 +1,13 @@
 var cdb = require('cartodb-deep-insights.js');
 var template = require('./widget-view.tpl');
 
+var widgetIconTemplateMap = {
+  category: require('./widget-icon-category.tpl'),
+  histogram: require('./widget-icon-histogram.tpl'),
+  formula: require('./widget-icon-formula.tpl'),
+  'time-slider': require('./widget-icon-timeSlider.tpl')
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js]: One more thing, time-slider is not the correct name, I've just checked. It is 'time-series', could you change them please? Sorry man :(.

Comment (piensaenpixel) [inline lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js]: Done :dancer:

File: lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js [hunk 5]
```diff
@@ -1,6 +1,13 @@
 var cdb = require('cartodb-deep-insights.js');
 var template = require('./widget-view.tpl');
 
+var widgetIconTemplateMap = {
+  category: require('./widget-icon-category.tpl'),
+  histogram: require('./widget-icon-histogram.tpl'),
+  formula: require('./widget-icon-formula.tpl'),
+  'time-series': require('./widget-icon-timeSlider.tpl')
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js]: And could you rename the template? :joy:

Comment (piensaenpixel) [inline lib/assets/javascripts/cartodb3/editor/widgets/widget-view.js]: pfff sorry

**Comment by xavijam** [comment]
:+1:

**Pull request merged**